src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -57,12 +57,40 @@ static ucs_status_t uct_cuda_ipc_rkey_unpack(uct_md_component_t *mdc,
     uct_cuda_ipc_key_t *key;
     ucs_status_t status;
     CUdevice cu_device;
+    CUdevice own_cu_device;
+    CUdevice peer_cu_device;
+    CUuuid own_uuid;
     int peer_accessble;
+    int device_count;
+    int i;
 
     UCT_CUDA_IPC_GET_DEVICE(cu_device);
 
+    status = UCT_CUDADRV_FUNC(cuDeviceGetCount(&device_count));
+    if (status != UCS_OK) {
+        return UCS_ERR_IO_ERROR;
+    }
+
+    peer_cu_device = -1;
+    for (i = 0; i < device_count; i++) {
+
+        status = UCT_CUDADRV_FUNC(cuDeviceGet(&own_cu_device, i));
+        if (status != UCS_OK) {
+            return UCS_ERR_IO_ERROR;
+        }
+        
+        status = UCT_CUDADRV_FUNC(cuDeviceGetUuid(&own_uuid, own_cu_device));
+        if (status != UCS_OK) {
+            return UCS_ERR_IO_ERROR;
+        }
+        
+        if (!memcmp((void *) &own_uuid, (void *) &packed->uuid, sizeof(packed->uuid))) {
+            peer_cu_device = i;
+        }
+    }
```

Comment:
@Akshay-Venkatesh is it make sense to have a static table of peer accessible capability instead of making many cuda calls in each unpack?

Reply:
Definitely makes sense to cache and have a table. I'll push changes.